File: api/src/endpoint/repository/get_repository_commit_diff.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryCommitDiffResource};

pub struct GetRepositoryCommitDiff;

impl Endpoint for GetRepositoryCommitDiff {
    const PATH: &'static str = "/repository/{owner}/{repo}/commits/{sha}/diff";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryCommitDiffRequest;
    type Response = GetRepositoryCommitDiffResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetRepositoryCommitDiffRequest {}

pub type GetRepositoryCommitDiffResponse = Vec<RepositoryCommitDiffResource>;

File: gitdot-api-ts/src/endpoint/repository/get-repository-commit-diff.ts
import { z } from "zod";

import { RepositoryCommitDiffResource } from "../../resource";

export const GetRepositoryCommitDiffRequest = z.object({});
export type GetRepositoryCommitDiffRequest = z.infer<
  typeof GetRepositoryCommitDiffRequest
>;

export const GetRepositoryCommitDiff = {
  path: "/repository/{owner}/{repo}/commits/{sha}/diff",
  method: "GET",
  request: GetRepositoryCommitDiffRequest,
  response: z.array(RepositoryCommitDiffResource),
} as const;
export type GetRepositoryCommitDiff = typeof GetRepositoryCommitDiff;
